### PR TITLE
feat: aggregator total gas paid metric

### PR DESCRIPTION
**Pull request by JulianVentura**
# Aggregator total gas paid metric

## Description

This PR adds aggregator total gas paid while responding to tasks metric to the grafana dashboard.

## How to test

1. Start all services locally, including the metrics
2. Start sending proofs
3. Go to the "System Data" dashboard, you should see a new panel showing the total amount of ETH paid by the aggregator, right down the gas price of the last batch.
4. Shut down the aggregator and start it over again
5. You should see that on new batch finished, the metric will keep accumulating on previous data.

## Type of change

Please delete options that are not relevant.

- [X] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Refactor

## Checklist

- [ ] “Hotfix” to `testnet`, everything else to `staging`
- [ ] Linked to Github Issue
- [ ] This change depends on code or research by an external entity
  - [ ] Acknowledgements were updated to give credit
- [ ] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] This change is an Optimization
  - [ ] Benchmarks added/run
- [ ] Has a known issue
  - [Link to the open issue addressing it]() 
- [ ] If your PR changes the Operator compatibility (Ex: Upgrade prover versions)
  - [ ] This PR adds compatibility for operator for both versions and do not change batcher/docs/examples
  - [ ] This PR updates batcher and docs/examples to the newer version. This requires the operator are already updated to be compatible

### CODE REVIEW

File: core/chainio/avs_writer.go [hunk 1]
```diff
@@ -204,10 +204,10 @@ func (w *AvsWriter) SendAggregatedResponse(batchIdentifierHash [32]byte, batchMe
 	return retry.RetryWithData(respondToTaskV2Func, retry.RespondToTaskV2())
 }
 
-// Calculates the transaction cost from the receipt and compares it with the batcher respondToTaskFeeLimit
-// if the tx cost was higher, then it means the aggregator has paid the difference for the batcher (txCost - respondToTaskFeeLimit) and so metrics are updated accordingly.
-// otherwise nothing is done.
-func (w *AvsWriter) checkIfAggregatorHadToPaidForBatcher(tx *types.Transaction, batchIdentifierHash [32]byte) {
+// Calculates the transaction cost from the receipt and updates the total amount paid by the aggregaot metric
```
Comment (PatStiles) [inline core/chainio/avs_writer.go]: small nit: `aggregaot` -> `aggregator`

**Comment by PatStiles** [comment]
Ran it locally and it observed the expected behavior

**Review by PatStiles — APPROVED** [review]

File: grafana/provisioning/dashboards/aligned/aggregator_batcher.json [hunk 2]
```diff
@@ -789,6 +789,77 @@
       "title": "Accumulated Aggregator Extra Cost Paid [ETH]",
       "type": "stat"
     },
+    {
+      "datasource": {
+        "type": "prometheus",
+        "uid": "prometheus"
+      },
+      "description": "Accumulated gas cost the aggregator paid for the batcher when the tx cost was higher than the respondToTaskFeeLimit.",
```
Comment (MarcosNicolau) [inline grafana/provisioning/dashboards/aligned/aggregator_batcher.json]:
```suggestion
      "description": "Accumulated gas cost in ETH the Aggregator paid when sending RespondToTask transactions.",
```

**Review by MarcosNicolau — APPROVED** [review]
Works as expected.

File: metrics/metrics.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ type Metrics struct {
 	aggregatorGasCostPaidForBatcherTotal   prometheus.Gauge
 	aggregatorNumTimesPaidForBatcher       prometheus.Counter
 	numBumpedGasPriceForAggregatedResponse prometheus.Counter
+	aggregatorGasCostPaidTotal             prometheus.Gauge
```
Comment (uri-99) [inline metrics/metrics.go]: this should be a counter

File: metrics/metrics.go [hunk 4]
```diff
@@ -113,6 +119,10 @@ func (m *Metrics) AddAggregatorGasPaidForBatcher(value float64) {
 	m.aggregatorGasCostPaidForBatcherTotal.Add(value)
 }
 
+func (m *Metrics) AddAggregatorGasCostPaidTotal(value float64) {
+	m.aggregatorGasCostPaidTotal.Add(value)
```
Comment (uri-99) [inline metrics/metrics.go:135]: yes, a counter.
gauges get "current data" , they can go up or down, like a pressure gauge of a steam machine.
a counter is for increments like this one, of values that count x occurrances, like km travelled with a car

**Review by uri-99 — CHANGES_REQUESTED** [review]

**Review by uri-99 — CHANGES_REQUESTED** [review]
Edit: removed comment

File: metrics/metrics.go (hunk 4 above)
Comment (JulianVentura) [reply metrics/metrics.go:135]: Yep, seems that's right. I'll do the change

**Review by uri-99 — APPROVED** [review]

**Review by JuArce — APPROVED** [review]

**Pull request merged**